lib/srv/alpnproxy/local_proxy.go

```diff
@@ -149,7 +150,8 @@ func (l *LocalProxy) Start(ctx context.Context) error {
 
 		if l.cfg.Middleware != nil {
 			if err := l.cfg.Middleware.OnNewConnection(ctx, l, conn); err != nil {
-				log.WithError(err).Errorf("Middleware failed to handle new connection.")
+				closeErr := conn.Close()
+				log.WithError(trace.NewAggregate(err, closeErr)).Errorf("Middleware failed to handle new connection.")
```

Comment:
Here's an example of what it looks like when I close the connection after it's already been closed:
```
time="2022-11-08T13:20:23-08:00" level=error msg="Middleware failed to handle new connection." error="access denied., close tcp 127.0.0.1:55473->127.0.0.1:55474: use of closed network connection"
```

I'll split this into separate logs since the close error is unrelated to the handler failing

Reply:
split version:
```
=== RUN   TestLocalProxyClosesConnOnError
time="2022-11-08T13:25:12-08:00" level=error msg="Middleware failed to handle new connection." error="access denied."
time="2022-11-08T13:25:12-08:00" level=error msg="Failed to close client connection." error="close tcp 127.0.0.1:55816->127.0.0.1:55817: use of closed network connection"
```